FinnUI/Sources/MyVehicles/MyVehiclesListView/MyVehiclesListView.swift

```diff
@@ -0,0 +1,118 @@
+import SwiftUI
+
+@available(iOS 13.0.0, *)
+struct MyVehiclesListView: View {
+
+    // MARK: - Private
+
+    @State var navigateToPreviouslyOwnedVehicles = false
+    private let imageUrl = "https://images.finncdn.no/dynamic/default/"
+
+    // MARK: - Public
+
+    var viewModel: MyVehiclesListViewModel
+    var onTapAddNewVehicle: (() -> Void) = {}
+
+    var body: some View {
+        NavigationView {
+            List {
+                addNewVehicle
+                ForEach(viewModel.vehicles, id: \.self) { viewModel in
+                    NavigationLink(destination: EmptyView()) {
+                        MyVehicleCell(
+                            viewModel: viewModel,
+                            imageProvider: SampleSingleImageProvider(
+                                url: self.constructURL(imagePath: viewModel.imagePath))
+                        )
+                    }
+                }
+                Group {
+                    seePreviouslyOwnedVehicles
+                    NavigationLink(destination: EmptyView(), isActive: $navigateToPreviouslyOwnedVehicles) {
+                        EmptyView()
+                    }
+                }
+            }
+            .appearance { (view: UITableView) in
+                view.separatorStyle = .none
+                view.backgroundColor = .bgPrimary
+            }
+            .appearance { (view: UITableViewCell) in
+                view.selectionStyle = .none
+            }
+            .navigationBarTitle("\(viewModel.title)", displayMode: .inline)
+        }
+        .navigationViewStyle(StackNavigationViewStyle())
+    }
+}
+
+@available(iOS 13.0.0, *)
+private extension MyVehiclesListView {
+    var addNewVehicle: some View {
+        HStack {
+            Image(.plus)
+                .renderingMode(.template)
+                .resizable()
+                .frame(width: .spacingL, height: .spacingL)
+                .foregroundColor(.btnPrimary)
+                .padding(.leading, .spacingM)
+
+            VStack(alignment: .leading) {
+                HStack(alignment: .center) {
+                    Text(viewModel.addNewVehicleTitle)
+                        .finnFont(.bodyStrong)
+                        .foregroundColor(.textAction)
+                        .padding(.leading, .spacingM)
+                    Image(.webview)
+                        .renderingMode(.template)
+                        .resizable()
+                        .frame(width: .spacingM, height: .spacingM)
+                        .foregroundColor(.iconPrimary)
+                }.padding(.top, 12)
+                divider.padding([.leading, .trailing], 12)
+            }
+        }.onTapGesture {
```

Comment:
this can be

```
.onTapGesture(perform: onTapAddNewVehicle)
```

maybe I would remove the tap on the method name, bad example, but if you change the actual gesture to the double tap you would have to rename the method.

Reply:
Good point!